### PR TITLE
HDDS-1580.Obtain Handler reference in ContainerScrubber

### PR DESCRIPTION
**Pull request by shwetayakkali**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 43 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 1 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 570 | trunk passed |
| +1 | compile | 286 | trunk passed |
| +1 | checkstyle | 80 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 934 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 152 | trunk passed |
| 0 | spotbugs | 311 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 513 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 491 | the patch passed |
| +1 | compile | 279 | the patch passed |
| +1 | javac | 279 | the patch passed |
| -0 | checkstyle | 42 | hadoop-hdds: The patch generated 3 new + 0 unchanged - 0 fixed = 3 total (was 0) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 748 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 155 | the patch passed |
| +1 | findbugs | 513 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 169 | hadoop-hdds in the patch failed. |
| -1 | unit | 1486 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 51 | The patch does not generate ASF License warnings. |
| | | 6732 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.TestMiniOzoneCluster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-842/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/842 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 766d657f4854 4.4.0-144-generic #170~14.04.1-Ubuntu SMP Mon Mar 18 15:02:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 77c49f2 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/1/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/1/testReport/ |
| Max. process+thread count | 4683 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service U: hadoop-hdds/container-service |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 36 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 543 | trunk passed |
| +1 | compile | 283 | trunk passed |
| +1 | checkstyle | 89 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 899 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 162 | trunk passed |
| 0 | spotbugs | 283 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 471 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 503 | the patch passed |
| +1 | compile | 273 | the patch passed |
| +1 | javac | 273 | the patch passed |
| +1 | checkstyle | 92 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 688 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 160 | the patch passed |
| +1 | findbugs | 497 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 170 | hadoop-hdds in the patch failed. |
| -1 | unit | 1491 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 55 | The patch does not generate ASF License warnings. |
| | | 6628 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.common.impl.TestContainerPersistence |
|   | hadoop.ozone.container.common.statemachine.commandhandler.TestBlockDeletion |
|   | hadoop.ozone.scm.node.TestQueryNode |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-842/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/842 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 1ca7a6a7d663 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 9c61494 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/2/testReport/ |
| Max. process+thread count | 5360 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/container-service U: hadoop-hdds/container-service |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-842/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by shwetayakkali** [comment]
The failed test TestHddsDatanodeService fails locally on trunk.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerScrubber.java [hunk 1]
```diff
@@ -130,12 +128,13 @@ private void throttleScrubber(TimeStamp startTime) {
...
   private void scrub() {
 
-    Iterator<Container> containerIt = containerSet.getContainerIterator();
+    Iterator<Container> containerIt = controller.getContainerSetIterator();
     long count = 0;
 
     while (containerIt.hasNext()) {
       TimeStamp startTime = new TimeStamp(System.currentTimeMillis());
       Container container = containerIt.next();
+      Handler containerHandler = controller.getHandler(container);
```
Comment (hanishakoneru) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerScrubber.java]: containerHandler is not used here. Is it for upcoming patches?

Comment (shwetayakkali) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerScrubber.java]: Yes @hanishakoneru it is for later use in the Directory scanner part.

**Comment by hanishakoneru** [comment]
Thank you @shwetayakkali . 
+1. I will merge it shortly.

**Pull request merged**